packages/react/src/useBatchPurchase.ts

```diff
@@ -0,0 +1,78 @@
+import { Signer } from '@ethersproject/abstract-signer'
+import { BigNumberish } from '@ethersproject/bignumber'
+import { TransactionHash, UUID } from '@liteflow/core'
+import { useCallback, useContext, useState } from 'react'
+import invariant from 'ts-invariant'
+import { LiteflowContext } from './context'
+import { ErrorMessages } from './errorMessages'
+
+export enum BatchPurchaseStep {
+  INITIAL,
+  TRANSACTION_SIGNATURE,
+  TRANSACTION_PENDING,
+  OWNERSHIP,
+}
+
+type batchPurchaseFn = (
+  purchases: { offerId: UUID; quantity: BigNumberish }[],
+) => Promise<void>
+
+export default function useBatchPurchase(signer: Signer | undefined): [
+  batchPurchaseFn,
+  {
+    activeStep: BatchPurchaseStep
+    transactionHash: TransactionHash | undefined
+  },
+] {
+  const { client } = useContext(LiteflowContext)
+  const [activeStep, setActiveProcess] = useState<BatchPurchaseStep>(
+    BatchPurchaseStep.INITIAL,
+  )
+  const [transactionHash, setTransactionHash] = useState<TransactionHash>()
+
+  const onProgress = useCallback(
+    (state) => {
+      switch (state.type) {
+        case 'TRANSACTION_SIGNATURE': {
+          setActiveProcess(BatchPurchaseStep.TRANSACTION_PENDING)
+          break
+        }
+        case 'TRANSACTION_PENDING': {
+          setActiveProcess(BatchPurchaseStep.TRANSACTION_PENDING)
+          setTransactionHash(state.payload.txHash)
+          break
+        }
+        case 'OWNERSHIP': {
+          setActiveProcess(BatchPurchaseStep.OWNERSHIP)
+          break
+        }
+        case 'OFFER_VALIDITY': {
+          console.log('validating offer...')
```

Comment:
console.log here.
could add a new `BatchPurchaseStep` or reuse an existing one if it make sense

Reply:
There is also a console.log in this other hook that I guess you start from: https://github.com/liteflow-labs/liteflow-js/blob/5ca797516fb96ad626d5461ee2869c977653756b/packages/react/src/useAcceptOffer.ts#L59

Reply:
Similar here, that's good, but it should be done for the whole library to include this step, adding it here without the rest will make things inconsistent.

Reply:
moved to a dedicated issue https://github.com/liteflow-labs/liteflow-js/issues/204